src/styles/main.scss

```diff
@@ -1,3 +1,62 @@
+@import 'variables/variables';
 body {
   margin: 0;
+  --grey: #eee;
+}
+
+@function calculate-margin-left($day) {
+  @return (index($days, $day) - 1) * $day-width
+    + 1px *(index($days, $day) - 1);
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-day;
+  width: ($day-width + $gap-day) * 7;
+  padding: 10px;
+  margin: auto;
+
+  &__day {
+    align-items: center;
+    box-sizing: border-box;
+    justify-content: center;
+    background-color: var(--grey);
+    width: $day-width;
+    height: 100px;
+    border: 1px solid black;
+    transition: all 0.5s;
+    display: none;
+
+    &::before {
+      font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
+      font-size: 30px;
+      text-align: center;
+      line-height: 50px;
+    }
+  }
+}
+
+@for $i from 1 through $days-in-month {
+  .calendar__day:nth-child(#{$i})::before {
+    content: "#{$i}";
+  }
+}
+
+@each $day in $days {
+  .calendar__day--#{$day} {
+    margin-left: calculate-margin-left(#{$day});
+  }
+}
+
+@for $length from 1 through $days-in-month {
+  .calendar__day:nth-child(#{$length}) {
+    display: flex;
```

Comment:
I believe this property is always static, you don't need a loop for that, 7'th point in checklist